index.html

```diff
@@ -679,6 +679,34 @@ <h2><dfn>readable</dfn> attribute</h2>
       with its length or to wait a defined length of time before transmitting
       the next message. Implementing a {{TransformStream}} for these types of
       message boundaries is left as an exercise for the reader.
+
+      <p>
+      While the {{ReadableStreamDefaultReader/read()}} method is asynchronous
+      and does not block execution, in code using async/await syntax it can
+      seem as if it does. In this situation it may be helpful to implement a
+      timeout which will allow the code to continue execution if no data is
+      received for a period of time. The example below uses the
+      {{ReadableStreamDefaultReader/releaseLock()}} method to interrupt a call
+      to {{ReadableStreamDefaultReader/read()}} after a timer expires. This
+      will not close the stream and so any data received after the timeout can
+      still be read later after calling {{ReadableStream/getReader()}} again.
+
+      <pre class="js">
+        async function readWithTimeout(port, timeout) {
```

Comment:
I thought it would be better to use `Promise.race()` but I'm not sure anymore. WDYT?
```js
async function readWithTimeout(port, timeout) {
  const reader = port.readable.getReader();

  const timerPromise = new Promise((resolve) => {
    setTimeout(resolve, timeout);
  });
  const readPromise = reader.read();

  const result = await Promise.race([readPromise, timerPromise]);
  reader.releaseLock();
  return result;
}
```

Reply:
That would also work but has slightly different behavior on timeout. My version rejects with the `TypeError` thrown by `read()` when `releaseLock()` is called while yours will resolve with `undefined`. Either is fine as long as the caller expects the right behavior.

I think the only motivation for one over the other is that your version allocates more promises and doesn't cancel the timer when it is no longer necessary. That said, my version doesn't cancel the timer if `read()` rejects for a different reason.

Reply:
I think yours is easier to understand.